src/hooks/useApi.ts

```diff
@@ -11,28 +12,15 @@ const BASE_URL = '/api';
  */
 const useApi = <T>(path: string): [Error | null, T | null] => {
   const [result, setResult] = useState<T | null>(null);
-  const [error, setError] = useState<Error | null>(null);
+  const callback = useCallback(async () => {
+    setResult(null);
+    setResult(await get(path));
```

Comment:
This looks a bit weird, but what it does is to set the result to `null` until the resources are actually fetched from the server. So the second `setResult()` is not called immediately, it's only called after `get()` successfully returns. The state in-between is a *loading* state.